packages/react-components/react-tree/src/components/Tree/useTree.ts

```diff
@@ -16,11 +16,20 @@ import { useControllableCheckedItems } from './useControllableCheckedItems';
 import { useTreeContext_unstable } from '../../contexts/treeContext';
 import { useRootTree } from '../../hooks/useRootTree';
 import { useSubtree } from '../../hooks/useSubtree';
+import { HTMLElementWalker, createHTMLElementWalker } from '../../utils/createHTMLElementWalker';
+import { treeItemFilter } from '../../utils/treeItemFilter';
 
 export const useTree_unstable = (props: TreeProps, ref: React.Ref<HTMLElement>): TreeState => {
   const [openItems, setOpenItems] = useControllableOpenItems(props);
   const [checkedItems] = useControllableCheckedItems(props);
-  const [navigate, navigationRef] = useTreeNavigation();
+  const { navigate, initialize } = useTreeNavigation();
+  const walkerRef = React.useRef<HTMLElementWalker>();
+  const initializeWalker = (root: HTMLElement | null) => {
```

Comment:
this needs to be wrapped in `useCallback`. Currently you will be creating a new HTMLElementWalker for every new rerender